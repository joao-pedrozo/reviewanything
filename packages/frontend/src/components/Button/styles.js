import styled, { css } from 'styled-components'

export const StyledButton = styled.button`
  height: 3.2rem;
  width: 26rem;
  font-size: 1.2rem;
  border-radius: 4px;
  font-weight: 700;
  ${({ secondary }) => css`
    background: ${secondary ? '#FFFFFF' : '#E74C3C'};
    color: ${secondary ? '#E74C3C' : '#FFFFFF'};
    border: ${secondary ? 'none' : '1px solid #DADADA'};
  `}
`