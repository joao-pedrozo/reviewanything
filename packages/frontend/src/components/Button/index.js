import React from 'react';

import { StyledButton } from './styles'

const Button = ({ 
  action, 
  secondary = false,
  children = 'Teste', 
  type = 'submit', 
}) => (
  <StyledButton 
    type={type} 
    action={action} 
    secondary={secondary}
    >
      {children}
    </StyledButton>
  )

export default Button