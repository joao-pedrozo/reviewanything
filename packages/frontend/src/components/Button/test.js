import React from 'react'
import { screen, render } from '@testing-library/react'
import { toHaveStyle } from '@testing-library/jest-dom'
import 'jest-styled-components'

import Button from '.'

describe('<Button />', () => {
  it('should render the default button', () => {
    const { container } = render(<Button>Entrar</Button>)
    expect(screen.getByRole('button', { name: /Entrar/i })).toHaveStyle({
      height: '3.2rem',
      width: '26rem',
      'font-size': '1.2rem',
      background: '#E74C3C',
      color: '#FFFFFF',
      'border-radius': '4px'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the secondary button version', () => {
    render(<Button secondary>Entrar</Button>)
    expect(screen.getByRole('button', { name: /Entrar/i })).toHaveStyle({
      height: '3.2rem',
      width: '26rem',
      'font-size': '1.2rem',
      color: '#E74C3C',
      background: '#FFFFFF',
      'border-radius': '4px'
    })
  })
})