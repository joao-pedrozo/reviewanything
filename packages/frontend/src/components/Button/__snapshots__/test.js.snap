// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the default button 1`] = `
.c0 {
  height: 3.2rem;
  width: 26rem;
  font-size: 1.2rem;
  border-radius: 4px;
  font-weight: 700;
  background: #E74C3C;
  color: #FFFFFF;
  border: 1px solid #DADADA;
}

<button
  class="c0"
  type="submit"
>
  Entrar
</button>
`;
